test/src/test/java/org/corfudb/integration/LogReplicationListenerIT.java

```diff
@@ -0,0 +1,820 @@
+        @Getter
+        private final List<CorfuStoreEntry<SampleSchema.Uuid, SampleSchema.SampleTableAMsg, SampleSchema.Uuid>>
+                existingEntries = new ArrayList<>();
+
+        LRTestListener(CorfuStore corfuStore, String namespace, CountDownLatch countDownLatch) {
+            super(corfuStore, namespace);
+            this.countDownLatch = countDownLatch;
+        }
+
+        @Override
+        public void onError(Throwable throwable) {
+            log.error("Error on listener", throwable);
+            fail("onError for LRTestListener: " + throwable.toString());
+        }
+
+        @Override
+        protected void onSnapshotSyncStart() {
+            log.info("Snapshot sync started");
+            countDownNumTxLatch();
+        }
+
+        @Override
+        protected void onSnapshotSyncComplete() {
+            log.info("Snapshot sync complete");
+            countDownNumTxLatch();
+        }
+
+        @Override
+        protected void processUpdatesInSnapshotSync(CorfuStreamEntries results) {
+            log.info("Processing updates in snapshot sync.");
+            updates.add(results);
+            results.getEntries().forEach((key, val) -> countDownLatch.countDown());
+            countDownNumTxLatch();
+        }
+
+        @Override
+        protected void processUpdatesInLogEntrySync(CorfuStreamEntries results) {
+            log.info("Processing updates in log entry sync.");
+            updates.add(results);
+            results.getEntries().forEach((key, val) -> countDownLatch.countDown());
+            countDownNumTxLatch();
+        }
+
+        @Override
+        protected void performFullSync(TxnContext txnContext) {
+            List<CorfuStoreEntry<SampleSchema.Uuid, SampleSchema.SampleTableAMsg, SampleSchema.Uuid>> entries =
+                    txnContext.executeQuery(userTableName, p -> true);
+            existingEntries.addAll(entries);
+
+            existingEntries.forEach(existingEntry -> countDownLatch.countDown());
+            countDownNumTxLatch();
+        }
+
+        private void countDownNumTxLatch() {
+            if (numTxLatch != null) {
+                numTxLatch.countDown();
+            }
+        }
+    }
+
+    /**
+     * This class is for LogicalGroup E2E IT
+     *
+     * (1) Open a table that store all the existing data
+     * (2) process updates, add the diff to the table and updates
+     */
+    private class MergeTestListener extends LogReplicationListener {
+
+        private final CountDownLatch updatesLatch;
+
+        private final List<String> tablesOfInterest;
+
+        private final CountDownLatch numTxLatch;
+
+        // Updates received through streaming
+        @Getter
+        private final List<CorfuStreamEntry> updates = new ArrayList<>();
+
+        // Entries discovered in performFullSync() before streaming updates are received
+        @Getter
+        private final List<CorfuStoreEntry>
+                existingEntries = new ArrayList<>();
```

Comment:
Didn't see how the diff is computed for existingEntries for snapshot sync updates, and how is the existingEntries consumed onSnapshotSyncComplete?